src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -968,24 +970,30 @@ func (c *controller) getProgressingOLMOperators() ([]*models.MonitoredOperator,
 
 func (c *controller) updatePendingOLMOperators(ctx context.Context) error {
 	c.log.Infof("Updating pending OLM operators")
-	operators, err := c.getProgressingOLMOperators()
+
+	// We don't want to use cache in this case as this function is executed once and we need to get up-to-date results.
```

Comment:
I understand that a function only executing once reduces the impact of not using cache.

But I think the primary motive is that we need up to date results. 

Does that sound fair?

Reply:
Why is it not important to get up to date results in the other locations?
Can we maybe add a comment in those locations to explain why we choose or do not choose cache for a given case. I think that this change is quite complex to read and understand so anything that aids clarification is good.